app/src/main/java/com/github/sdp/mediato/model/User.java

```diff
@@ -0,0 +1,127 @@
+package com.github.sdp.mediato.model;
+
+import com.github.sdp.mediato.errorCheck.Preconditions;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class User {
+    public static final int LIMIT_LOCATION_SIZE = 2;
+    public static final String FAVOURITES_COLLECTION = "Favourites";
+    public static final String RECENTLY_WATCHED_COLLECTION = "Recently Watched";
+
+    private String id = "";
+    private String username = "";
+    private String displayedName = "";
+    private String email = "";
+    private String birthDate = "";
+    private String registerDate = "";
+
+    private List<Double> location = new ArrayList<>();
+    private List<User> friends = new ArrayList<>();
+    private Map<String, List<Review>> reviews = new HashMap<>();
+    private User(){}
+    public User(UserBuilder builder){
+        this.id = builder.id;
+        this.username = builder.username;
+        this.displayedName = builder.displayedName;
+        this.email = builder.email;
+        this.birthDate = builder.birthDate;
+        this.registerDate = builder.registerDate;
+        this.location = builder.location;
+    }
+
+    public String getId() {
+        return id;
+    }
+
+    public String getUsername() {
+        return username;
+    }
+
+    public String getDisplayedName() {
+        return displayedName;
+    }
+
+    public String getEmail() {
+        return email;
+    }
+
+    public String getBirthDate() {
+        return birthDate;
+    }
+
+    public String getRegisterDate() {
+        return registerDate;
+    }
+
+    public List<Double> getLocation() {
+        return location;
+    }
+
+    public List<User> getFriends() {
+        return friends;
+    }
+
+    public static class UserBuilder {
+        private String id = "";
+        private String username = "";
+        private String displayedName = "";
+        private String email = "";
+        private String birthDate = "";
+        private String registerDate = "";
+
+        private List<Double> location = new ArrayList<>();
+        private List<User> friends = new ArrayList<>();
+        private Map<String, List<Review>> reviews = new HashMap<>();
+
+        public UserBuilder(String id){

```

Comment:
Does this mean that only the String id is a required field to build a user? I think there should be more such as username and email.

Reply:
All the fields except friends and reviews are "mandatory" (checked before the build() completes). I chose this model since it allows us to set the different fields from different fragments if we need to.